gammapy/maps/wcsnd.py

```diff
@@ -589,6 +585,63 @@ def get_spectrum(self, region=None, func=np.nansum, weights=None):
 
         return self.to_region_nd_map(region=region, func=func, weights=weights)
 
+    def binary_erosion(self, margin, from_edges=False):
+        """Binary erosion of boolean mask removing a given margin
+        Parameters
+        ----------
+        margin : tuple of `~astropy.units.Quantity`
+            Angular sizes of the margin in (lon, lat) in that specific order.
+            If only one value is passed, the same margin is applied in (lon, lat).
+        from_edge : bool
+            If True apply the margin only from edges (Default is False).
+        """
+        if self.data.dtype != bool:
+            raise (TypeError, "Binary erosion should be applied only on boolean mask")
+        if from_edges:
+            mask_margin = np.ones(self.data.shape[-2:], dtype=bool)
+        else:
+            # TODO: should we apply the erosin independently to each other dim
+            # for now just combine non-spatial dim with OR condition
+            axis = tuple(np.arange(len(self.geom.axes)))
+            mask_margin = np.sum(self.data, axis=axis).astype(bool)
+        if not isinstance(margin, tuple):
+            margin = (margin, margin)
+        shape = tuple(
+            [
+                int(np.ceil(x / scale).value * 2 + 1)
+                for x, scale in zip(margin, self.geom.pixel_scales)
+            ]
+        )
+        mask_margin = ndi.binary_erosion(
+            mask_margin[..., :, :], structure=np.ones(shape)
+        )
+        idx_y, idx_x = np.where(~mask_margin)
+        self.data[..., idx_y, idx_x] = False
```

Comment:
I think we should return a new map here just as we do for `.smooth()` and `.convolve()`...